server/datastore/mysql/hosts.go

```diff
@@ -1881,8 +1884,18 @@ func (ds *Datastore) UpdateHost(ctx context.Context, host *fleet.Host) error {
 	return nil
 }
 
-// OSVersions gets the aggregated os version host counts. If a non-nil teamID is passed, it will filter hosts by team.
-func (ds *Datastore) OSVersions(ctx context.Context, teamID *uint, platform *string, osID *uint) (*fleet.OSVersions, error) {
+// OSVersions gets the aggregated os version host counts. Records with the same name and version are combined into one count (e.g.,
+// counts for the same macOS version on x86_64 and arm64 architectures are counted together.
+// Results can be filtered using the following optional criteria: team id, platform, or name and
+// version. Name cannot be used without version, and conversely, version cannot be used without name.
+func (ds *Datastore) OSVersions(ctx context.Context, teamID *uint, platform *string, name *string, version *string) (*fleet.OSVersions, error) {
+	if name != nil && version == nil {
```

Comment:
Seems like we are duplicating the logic for validating parameters in server/service/hosts.go. Do we need to do this here?

Reply:
@michalnicp, I added these here because I saw this being called directly in parts of the code so thought validating here as well would be wise.